dt: Adding rpi-cn0554 support

catapangan opened the pull request:
Signed-off-by: Angelo Catapang <Angelo.Catapang@analog.com>

nunojsa commented on arch/arm/boot/dts/overlays/rpi-cn0554-overlay.dts:
```diff
@@ -0,0 +1,248 @@
+		__overlay__ {
+			status = "disabled";
+		};
+	};
```
Disabling the spidevs should come before the ad7124 and ltc2688 nodes.

nunojsa reviewed (commented):
Some notes on this one:

1) You need to SPDX tags `// SPDX-License-Identifier: GPL-2.0` together with ADI Copyright (see other overlays)
2) Please use new overlay syntax. Metawords as `fragment` should not be used. Look at the [ltc2688](https://github.com/analogdevicesinc/linux/blob/rpi-5.10.y/arch/arm/boot/dts/overlays/rpi-ltc2688-overlay.dts) overlay for an example of the new syntax.
3) Please add a small description on the commit message on what is being added (moreover if we need the extra default overlay). The commit title should also have the form `arch: arm: dts: add cn0554 overlay`

Also, do we really need both overlays? Can we  somehow use `__overrides__` so that we can have only one overlay?

bbushey commented:
Hi Nuno,
The LCT2688 overlay really is specific to that eval board.  It doesn't make a ton of sense to add the AD7124 into the LTC2688 overlay, and the CN0554 board has hardware specific ranges and channel configurations that need to be specified.  So, with all this said we think having a separate overlay for the CN0554 board is really the correct way to scale this properly.

nunojsa commented:
Hi Brandon,

> The LCT2688 overlay really is specific to that eval board. It doesn't make a ton of sense to add the AD7124 into the LTC2688 overlay,

Not sure if I'm following :). From what I can see, the CN0554 default overlay also has the ad7124 defined and for the ltc26888 standalone, there's already one overlay.

From what I can see, only the configuration of the ltc2688 channels is changing between both overlays which means we could, most likely, tweak the channels configuration using `__overrrides__`. Yes, the /boot/config.txt would look a bit odd but we would only need to maintain one overlay.

That said, up to you guys to keep (or not) both overlays...

mthoren-adi commented:
Hi all, indeed - the CN0554 overlay should include both the AD7124 and LTC2688 devices.
@catapangan can you just commit one overlay for now - the "default" one is correct, but let's not call it "default" - let's commit a single rpi-cn0554-overlay.dts, and add additional ones later, or perhaps add overrides to this one once we figure out a clean way of doing so.

catapangan commented:
Hi @mthoren-adi , @nunojsa 

Changed the commit changes to only one overlay rpi-cn0554-overlay.dts

Also, added the ff changes on the dts:
- Added SPDX tags
- Followed new syntax from ltc2688 dts
- Changed commit title

mthoren-adi commented:
@catapangan I'm able to apply the patch and build the dts. I did get these warnings, not sure if they're from your dts or the patch formatting:
analog@experimenter:~/git/linux $ git am 2110.patch
Applying: arch: arm: dts: add cn0554 overlay
.git/rebase-apply/patch:63: trailing whitespace.
.git/rebase-apply/patch:96: trailing whitespace.
.git/rebase-apply/patch:99: trailing whitespace.
.git/rebase-apply/patch:102: trailing whitespace.
.git/rebase-apply/patch:105: trailing whitespace.
warning: squelched 6 whitespace errors
warning: 11 lines add whitespace errors.

But it built fine:

analog@experimenter:~/git/linux $ make dtbs
  DTCO    arch/arm/boot/dts/overlays/rpi-cn0554.dtbo
analog@experimenter:~/git/linux $ 

and it got it to load on a RPi 4 w/ CN0544.

I'm still trying to reconcile the boards I have - I have two initial prototypes that did not have the AD7124 interrupt connected (which I connected via a jumper) and I THINK two of the second prototypes. So far I'm getting some response from the AD7124 but can't write to the LTC2688.

But if you can confirm that the hardware works on your end, and @nunojsa  is good with your other fixes, then I think we're good to go. (I'll try a few more things here, there's a chance all of my boards are compromised.)

nunojsa commented on arch/arm/boot/dts/overlays/rpi-cn0554-overlay.dts:
```diff
@@ -0,0 +1,216 @@
+/*
+ * Overlay for the CN0554 Mixed Signal Hat
+ *
+ * Copyright 2022 Analog Devices Inc.
```
2023?

nunojsa commented on arch/arm/boot/dts/overlays/rpi-cn0554-overlay.dts:
```diff
@@ -0,0 +1,216 @@
+			adi,output-range-microvolt = <(-15000000) 15000000>;
+		};
+	};
+};
```
you need to end with a new line...

catapangan commented:
Hi @nunojsa

Changed to unix line endings and removed trailing white spaces.

nunojsa commented:
> Changed to unix line endings and removed trailing white spaces.

It does not look like you did :). See...

https://dev.azure.com/AnalogDevices/OpenSource/_build/results?buildId=22473&view=logs&j=4df93447-0a2c-583c-0285-f584685adda3&t=5bbcc8da-92bd-587e-82f2-7fe5aa20e63d

NOTE: Assuming you have your branch rebased with rpi-5.10.y, you can run (from your kernel directory) `scripts/checkpatch.pl --git --strict origin/rpi-5.10.y..` and make sure everything is fine before pushing..

Also, squash the second commit into the first one...

LaiVietHai commented:
@nunojsa can you help me build the dtbo file, i can't build with dtc it always shows error rpi-cn0554-overlay.dts:11.1-9 syntax error.
Thanks you,
HaiLV
